cosmos/precompile/erc20/helpers.go

```diff
@@ -0,0 +1,159 @@
+// SPDX-License-Identifier: BUSL-1.1
+//
+// Copyright (C) 2023, Berachain Foundation. All rights reserved.
+// Use of this software is govered by the Business Source License included
+// in the LICENSE file of this repository and at www.mariadb.com/bsl11.
+//
+// ANY USE OF THE LICENSED WORK IN VIOLATION OF THIS LICENSE WILL AUTOMATICALLY
+// TERMINATE YOUR RIGHTS UNDER THIS LICENSE FOR THE CURRENT AND ALL OTHER
+// VERSIONS OF THE LICENSED WORK.
+//
+// THIS LICENSE DOES NOT GRANT YOU ANY RIGHT IN ANY TRADEMARK OR LOGO OF
+// LICENSOR OR ITS AFFILIATES (PROVIDED THAT YOU MAY USE A TRADEMARK OR LOGO OF
+// LICENSOR AS EXPRESSLY REQUIRED BY THIS LICENSE).
+//
+// TO THE EXTENT PERMITTED BY APPLICABLE LAW, THE LICENSED WORK IS PROVIDED ON
+// AN “AS IS” BASIS. LICENSOR HEREBY DISCLAIMS ALL WARRANTIES AND CONDITIONS,
+// EXPRESS OR IMPLIED, INCLUDING (WITHOUT LIMITATION) WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, NON-INFRINGEMENT, AND
+// TITLE.
+
+package erc20
+
+import (
+	"math/big"
+
+	"github.com/holiman/uint256"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	pbindings "pkg.berachain.dev/polaris/contracts/bindings/polaris"
+	"pkg.berachain.dev/polaris/eth/common"
+	ethprecompile "pkg.berachain.dev/polaris/eth/core/precompile"
+	"pkg.berachain.dev/polaris/eth/core/vm"
+)
+
+// salt is used to generate unique contract addresses for each ERC20 token.
+var salt uint64
+
+// deployPolarisERC20Contract deploys a new ERC20 token contract by calling back into the EVM.
+func (c *Contract) deployPolarisERC20Contract(
+	ctx sdk.Context,
+	evm ethprecompile.EVM,
+	deployer common.Address,
+	name string,
+	endowment *big.Int,
+) (common.Address, error) {
+	plugin := c.GetPlugin()
+	plugin.EnableReentrancy(ctx)
+	defer plugin.DisableReentrancy(ctx)
+
+	// deploy new ERC20 token contract
+	code := common.FromHex(pbindings.PolarisERC20MetaData.Bin)
+	args, err := c.polarisERC20ABI.Pack("", name, name)
+	if err != nil {
+		return common.Address{}, err
+	}
+	suppliedGas := ctx.GasMeter().GasRemaining()
+	_, contractAddr, gasRemaining, err := evm.Create2(
+		vm.AccountRef(deployer), append(code, args...), suppliedGas, endowment, uint256.NewInt(salt),
+	)
+	salt++
```

Comment:
I don't think this salt++ is doing anything.

Reply:
wdym? its a pretty common way to salt, just use a nonce/counter so it guarantees every salt is unique